### PR TITLE
Improve `kedro viz build` usage documentation 

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves https://github.com/kedro-org/kedro-viz/issues/1918

This PR adds additional documentation to clarify why `kedro-viz` cannot be run directly from a local filesystem (`file:// `protocol) without encountering specific issues. The intention is to provide more context to users who may try to use the file protocol and run into these issues.

## Context

Currently, attempting to run `kedro-viz` by simply opening index.html through the `file://` protocol causes [CORS issues](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSRequestNotHttp?utm_source=devtools&utm_medium=firefox-cors-errors&utm_campaign=default#loading_a_local_file). Most modern browsers enforce strict CORS policies for local files to prevent security vulnerabilities.

The file protocol (`file://`) is treated differently compared to other network requests. Requests to resources from a local file context (e.g., scripts, APIs) are treated as cross-origin requests by the browser, which results in CORS errors when trying to access JavaScript assets or other dynamic resources required by kedro-viz.

Therefore, the recommendation for a proper user experience is to run kedro-viz using a local web server, which avoids CORS policy issues and guarantees that all resources are correctly served and accessed. Examples of lightweight static servers include Python's `http.server` or Node.js `http-server`, which can easily serve the generated HTML.

This PR proposes updated documentation to explain why users need to run Kedro-Viz through a server, helping prevent confusion.

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by rashidakanchwala** [comment]
Quick question here - can we enable kedro-viz to run on a simple python using kedro viz cli locally ... maybe we could kedro viz <build path>

**Comment by jitu5** [comment]
> Quick question here - can we enable kedro-viz to run on a simple python using kedro viz cli locally ... maybe we could kedro viz

@rashidakanchwala If I understood correctly, then It will become `kedro viz run`  ?

**Comment by rashidakanchwala** [comment]
Sorry, I didn't read your PR properly. I meat running `python m http.server` under the hood, and some CLI command that executes it but I think this is good for now :)

File: docs/source/platform_agnostic_sharing_with_kedro_viz.md [hunk 1]
```diff
@@ -46,6 +46,32 @@ Starting from Kedro-Viz 9.2.0, `kedro viz build` will not include dataset previe
 
 This creates a `build` folder containing your Kedro-Viz app package in your project directory. 
 
+## Running Kedro-Viz Locally
+
+When you generate build folder with the command `kedro viz build`, it creates a build directory with an `index.html` file, which is the entry point for visualizing your pipeline. However, simply opening `index.html` using the file protocol `file://` is not supported due to Cross-Origin Resource Sharing Policy in modern browsers.
```
Comment (rashidakanchwala) [inline docs/source/platform_agnostic_sharing_with_kedro_viz.md:51]:
```suggestion
When you generate the build folder using the command `kedro viz build`, it creates a build directory with an `index.html` file, which serves as the entry point for visualizing your pipeline. However, simply opening `index.html` using the `file://` protocol is not supported due to Cross-Origin Resource Sharing (CORS) policies in modern browsers..
```

**Review by rashidakanchwala — APPROVED** [review]
Awesome, thanks for this!

**Review by astrojuanlu — COMMENTED** [review]
Thanks a lot @jitu5 ! Maybe I'd remove the "using a local server" sub-heading and rearrange the information like

"""
When you generate the build folder using the command `kedro viz build`, it creates a build directory with an `index.html` file, which serves as the entry point for visualizing your pipeline.

To view your pipeline visualization, you need to serve `index.html` using an HTTP server. Here are a few simple ways to do this:

(explain two different ways)

::: warning
simply opening `index.html` using the `file://` protocol is not supported due to Cross-Origin Resource Sharing (CORS) policies in modern browsers.
:::
"""

**Review by astrojuanlu — APPROVED** [review]
Thanks!

**Pull request merged**